[ADD] ecommerce: new configuration pages + Products page update (#16285)
odoo/documentation: content/applications/websites/ecommerce/configuration/products.rst

========
Products
========

Odoo eCommerce allows you to :ref:`create products <ecommerce/products/product-creation>` and
manage their :ref:`visibility <ecommerce/products/publish-products>`, offering a range of

> customization options. Products can be :ref:`configured <ecommerce/products/product-configuration>`

Reviewer: ```suggestion
manage their :ref:`visibility <ecommerce/products/publish-products>` on the website, offering a range of customization options. Products can be :ref:`configured <ecommerce/products/product-configuration>`
```
I'm also not sure what you are referring to here when you say *offering a range of customization options*? Could we actually get rid of that part of the sentence?

by adding :ref:`product images and videos <ecommerce/products/images>`, creating :ref:`product
combinations <ecommerce/products/combos>`, implementing :ref:`cross-selling and upselling strategies
<ecommerce/products/cross_upselling>`, attaching :ref:`digital files
<ecommerce/products/digital-files>`, :ref:`translating <ecommerce/products/translation>`
product content, using :ref:`stock management tools <ecommerce/products/stock-management>`, and

> selecting :ref:`packagings <ecommerce/products/stock-management/packagings>`.

Reviewer: and for this part I'm not sure it totally makes sense to say *Products can be configured by xxx* in this case because what we actually mean is *products can be configured AND we can add images, create product combos, implement ... etc.* Could we maybe think of of a better phrasing? Maybe something along these lines:

Products can be configured in various ways, including adding images and videos, creating product combinations, implementing cross-selling and upselling strategies, attaching digital files, translating product names and descriptions, managing stock, and setting packaging options.

the content.

.. tip::
   - Click the :icon:`fa-globe` :guilabel:`Go to Website` smart button to access the frontend
     product page and :doc:`customize <../ecommerce_design/product_page>` it using the :doc:`website
     editor <../../website/web_design>`.
   - Product descriptions can also be generated using :doc:`AI </applications/productivity/ai>`. To
     do so, type `/` in the :guilabel:`Ecommerce description` section and select :guilabel:`AI` to
     open the chatbot window. Provide instructions for the desired description, then click
     :guilabel:`Use this` to apply it. The `/AI` option is available in various fields and sections.

.. _ecommerce/products/images:

Product images and videos
-------------------------

To add media items, such as images and videos, navigate to the :ref:`product form
<ecommerce/products/product-form>` and go to the :guilabel:`Sales` tab. In the :guilabel:`Ecommerce
Media` section, click :guilabel:`Add Media`. In the :guilabel:`Select a media` pop-up window:

- To add an image, go to the :guilabel:`Images` tab, select an image, click :guilabel:`Add URL` or
  :guilabel:`Upload an image`.

> - To add a video, navigate to the :guilabel:`Videos` tab, paste a video URL or embed code.

Author: I found this: https://www.odoo.com/documentation/19.0/applications/websites/website/web_design/building_blocks.html#embed-code
You just need the HTML code of the video and insert it (as far as I understand). But where to get this piece of code depends on the source of the video. So, hence, I would maybe add "or embed HTML code" and the ref to the doc, but nothing more.

BTW I just see these options. Do you think we need to mention/explain them?
<img width="568" height="364" alt="Image" src="https://github.com/user-attachments/assets/5afe1e08-8870-4b29-a0b3-b671ab3d9fea" />

      .. image:: products/accessory-products.png
         :alt: Suggested accessories at checkout during cart review

      .. note::
         To hide this section, open the :doc:`website editor <../../website/web_design>`, go to the
         :guilabel:`Style` tab, and toggle the :guilabel:`Suggested Accessories` switch off.

   .. tab:: Alternative products

      Alternative products are displayed at the bottom of the :doc:`product page
      <../ecommerce_design/product_page>` under the :guilabel:`Alternative Products` heading.

      .. image:: products/alternative-products-ecommerce.png
         :alt: The alternative products section of an product page.

      .. tip::
         To customize this block, open the :doc:`website editor <../../website/web_design>` and
         select the related :doc:`building block <../../website/web_design/building_blocks>`. In the
         :guilabel:`Style` tab, scroll to the :guilabel:`Alternative Products` section and modify
         the settings as needed.

.. _ecommerce/products/digital-files:

Digital files
-------------

It is possible to link digital files, such as certificates, eBooks, or user manuals, to the

> products. To link a digital file to a product, go to the :ref:`product form

Reviewer: ```suggestion
products. To do so, go to the :ref:`product form
```
to avoid repetitions

<ecommerce/products/product-form>` and click the :icon:`fa-file-text-o` :guilabel:`Documents` smart
button. Then click :guilabel:`Upload` to upload a file directly, or for additional options, click
:guilabel:`New`. Choose the :guilabel:`Type` of attachment:

- :guilabel:`File`: :guilabel:`Upload your file`.
- :guilabel:`URL`: Insert the link to the file or media item.
- :guilabel:`Cloud Storage` (if applicable): Insert a link to your cloud storage.

These documents can be made available:

- On the product page (before checkout): Enable the :guilabel:`Publish on website` option on the
  document form or card. Set the :guilabel:`Sales visibility` field to :guilabel:`Hidden`.
- In the :doc:`customer portal </applications/general/users/user_portals>` on the confirmed sales
  order (after checkout): Set the :guilabel:`Sales visibility` field to :guilabel:`On confirmed

>   order` and turn off the :guilabel:`Publish on website` switch.

Reviewer: ```suggestion
  order` and disable :guilabel:`Publish on website`.
```
since when you're in the form view, this option is a check box and not a switch 😉

<ecommerce/products/additional_features/product-highlight>` names, :doc:`categories and variants
<categories_variants>` names, etc.

.. note::
   Having untranslated content on a web page may be detrimental to the user experience and
   :doc:`SEO </applications/websites/website/structure/seo>`. To avoid this, use the
   :ref:`Translate <translate/translate>` feature to translate the page's content.

.. seealso::
   :doc:`Website translations </applications/websites/website/configuration/translate>`

.. _ecommerce/products/stock-management:

Stock management
----------------

.. important::
   The :doc:`Inventory app </applications/inventory_and_mrp/inventory>` must be installed to handle
   stock-related settings and operations.

To ensure that e-commerce sales stay aligned with the inventory available in the warehouse,
configure the inventory management options. To do so, navigate to go to :menuselection:`Website -->
Configuration --> Settings`, scroll down to the :guilabel:`eCommerce` section, then to the
:guilabel:`Inventory Defaults` subsection. Configure the following stock-related options, if needed:

> 

Reviewer: Right! I guess we could mention that first when describing the field (e.g., "If multiple warehouses exist, select the relevant one; xxxx" --> this isn't great but then you can understand what I mean this way :slightly_smiling_face: )

.. note::
   - Enabling/disabling the general :guilabel:`Inventory Defaults` settings does not automatically
     (un)tick checkboxes on the product form for existing products; it only affects new products
     created after the feature is turned on/off.
   - To use the stock-related features under the :guilabel:`Sales` tab of the product form and to
     have the notification option available, the :ref:`Track inventory setting
     <inventory/product_management/tracking-inventory>` must be enabled on the product form.
   - To display the stock level on the product page, the :guilabel:`Product Type` field on the
     :ref:`product form <ecommerce/products/product-form>` must be set to :guilabel:`Goods` or
     :guilabel:`Combo`.
   - A :icon:`fa-envelope-o` (:guilabel:`envelope`) :guilabel:`Get notified when back in stock`
     button appears on the product page when an item is out of stock. Customers can click
     the link to enter their email address and receive a notification once the item is back
     in stock.
   - Use the :ref:`Click & Collect <ecommerce/shipping/instore-pickup>` feature to display the
     product availability on the product page.

> .. example::

Reviewer: I'm sorry, I know I'm annoying, but I think maybe a better example would be one where the product is available for delivery but not click&collect, or vice versa, especially since email notifications and out-of-stock messages are IMO more simple features, whereas stock/warehouse management is a bit more complex; I actually struggled a bit myself with this and I think it would make for a more interesting example?

Author: I intentionally added an example of what it looks like when something is out of stock because we speak of 3 features that appear when it is out of stock. And I thought it was confusing without an image. 
 
But then, I am not sure whether I would add an example at all in this case. Because we are mostly explaining the out-of-stock features and not explicitly mentioning click and collect here, we just mention it. I don't know if it doesn't come a bit out of the blue because we don't explain how to configure it (as we do that in Delivery).

   Currently, the `Boko Chair` is not available in any warehouses. However, customers can click
   the :icon:`fa-envelope-o` (:guilabel:`envelope`) :guilabel:`Get notified when back in
   stock` link to receive a notification as soon as the product becomes available again. An
   :guilabel:`Out-of-Stock Message` can also be added to inform customers that the item will be

>    replenished.

Reviewer: what happens if the product is only available in one of the warehouses? is the get notified button displayed then? (i suppose not)

<ecommerce/products/cross_upselling>`, attaching :ref:`digital files
<ecommerce/products/digital-files>`, :ref:`translating <ecommerce/products/translation>`
product information, :ref:`managing stock <ecommerce/products/stock-management>`, and
setting :ref:`packaging options <ecommerce/products/stock-management/packagings>`.

.. _ecommerce/products/product-creation:

Product creation
================

To create a product from the frontend, go to the Website app, click :guilabel:`New`
:icon:`fa-caret-down` in the top-right corner, then select :guilabel:`Product`. Enter the
:guilabel:`Product Name`, and other relevant details such as the :guilabel:`Barcode`,
:guilabel:`Sales Price`, default :guilabel:`Sales Taxes`, and a :ref:`Website Category
<ecommerce/categories_variants/categories>`, then add an image. Once you click :guilabel:`Save`, you
are redirected to the product page where you can :ref:`customize it
<ecommerce_design/product_page/customization>` using the :doc:`website editor
</applications/websites/website/web_design>`. Once you are finished, click :guilabel:`Save`. To
configure additional advanced :ref:`product details <ecommerce/products/product-configuration>`,

> click the :guilabel:`Product` button in the top-right corner to access the backend product form.

Reviewer: I'm not sure I would call this a button per say:

```suggestion
configure additional advanced :ref:`product details <ecommerce/products/product-configuration>`,
click :guilabel:`Product` in the top-right corner to access the backend product form.
```
